php/libraries/Utility.class.inc

```diff
@@ -742,42 +758,14 @@ class Utility
      * @return array of study visits in the format array('VL' => 'VL')
      *         where VL is the visit label
      *
-     * @note       THIS FUNCTION USES STRING CONCATENATION TO GENERATE AN
-     *             SQL STATEMENT.
-     *             IT IS VULNERABLE TO AN SQL INJECTION ATTACK.
-     * @note       Function comment written by Dave, not the author of this function.
-     * @vulnerable
-     * @cleanup
+     * @deprecated
      */
     static function getExistingVisitLabels(int $projectID = null): array
     {
-        $factory = \NDB_Factory::singleton();
-        $db      = $factory->database();
-        $qparams = array();
-
-        if ($projectID != null) {
-            $ExtraProject_Criteria
-                = ' AND (s.ProjectID IS NULL OR s.ProjectID=:ProjectID)';
-            $qparams["ProjectID"]  = $projectID;
-        } else {
-            $ExtraProject_Criteria ='';
-        }
-        $query          = "SELECT DISTINCT Visit_label FROM session s
-            JOIN candidate c ON (s.CandID = c.Candid)
-            JOIN psc ON (s.CenterID = psc.CenterID)
-            WHERE c.Active = 'Y'
-            AND s.Active = 'Y'
-            AND c.Entity_type != 'Scanner'
-            AND psc.CenterID!= '1'
-            $ExtraProject_Criteria ORDER BY Visit_label";
-        $visitlabels    = $db->pselect($query, $qparams);
-        $visits         = array();
-        $numVisitLabels = count($visitlabels);
-        for ($i = 0; $i < $numVisitLabels; $i++) {
-            $visits[$visitlabels[$i]['Visit_label']]
-                = ($visitlabels[$i]['Visit_label']);
-        }
-        return $visits;
+        throw new \DeprecatedException(
```

Comment:
If there isn't any part of the code using this function, do we even need to keep it here to display this error message?

Reply:
I think we do for the case where a project is using it in their custom code.